source/patterns/@aws-solutions-constructs/aws-lambda-kendra/README.md

```diff
@@ -0,0 +1,133 @@
+
+## Overview
+
+This AWS Solutions Construct implements an AWS Lambda function and Amazon Kendra index with the least privileged permissions.
+
+Here is a minimal deployable pattern definition:
+
+Typescript
+
+```typescript
+import { Construct } from 'constructs';
+import { Stack, StackProps, Aws } from 'aws-cdk-lib';
+import { LambdaToKendra } from '@aws-solutions-constructs/aws-lambda-kendra';
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as s3 from "aws-cdk-lib/aws-s3";
+
+const lambdaProps: lambda.FunctionProps = {
+  code: lambda.Code.fromAsset(`lambda`),
+  runtime: lambda.Runtime.NODEJS_18_X,
+  handler: 'index.handler'
+};
+
+new LambdaToKendra(this, 'sample', {
+  lambdaFunctionProps: lambdaProps,
+  kendraIndexProps: {},
+  kendraDataSourceProps: [{
+    type: 'S3',
+    dataSourceConfiguration: {
+      s3Configuration: {
+        bucketName: 'your-bucket-name',
+      }
+    }
+  ],
+});
+```
+
+Python
+
+```python
+TBD
+```
+
+Java
+
+```java
+TBD
+```
+
+## Pattern Construct Props
+
+| **Name** | **Type** | **Description** |
+| --- | --- | --- |
+| existingLambdaObj? | [`lambda.Function`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.Function.html) | Existing instance of Lambda Function object, providing both this and `lambdaFunctionProps` will cause an error. |
+| lambdaFunctionProps? | [`lambda.FunctionProps`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_lambda.FunctionProps.html) | User provided props to override the default props for the Lambda function. |
+| kendraIndexProps? | [`kendra.CfnIndexProps`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_kendra.CfnIndex.html) | Optional user provided props to override the default props for the Kendra index. **Is this required?** |
+| kendraDataSourcesProps | [`CfnDataSourceProps[]`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_kendra.CfnDataSource.html) | A list of data sources that will provide data to the Kendra index. *?At least 1 must be specified*. We will do majority of processing for some data sources (S3 crawler initially), but for others the props must be complete (e.g. proper roleArn, etc.) |
+| indexPermissions? | `string[]` | Optional - index permissions to grant to the Lambda function. One or more of the following may be specified: `Read`, `SubmitFeedback` and `Write`. Default is `["Read", "SubmitFeedback"]`. Read is all the operations IAM defines as Read and List. SubmitFeedback is only the SubmitFeedback action. Write is all the operations IAM defines as Write and Tag. This functionality may be overridden by providing a specific role arn in lambdaFunctionProps |
+| existingKendraIndex | [`kendra.cfnIndex`](https://docs.aws.amazon.com/cdk/api/v2/docs/aws-cdk-lib.aws_kendra.CfnIndex.html) | An existing Kendra index to which the Lambda function will be granted access. Supplying along with kendraIndexProps or kendraDataSourceProps will throw an error. |
```

Comment:
In the `index.ts` file, I believe this property is called `existingKendraIndexObj` and is optional, so we should update the README to reflect it.